kotest-property/src/jvmTest/kotlin/com/sksamuel/kotest/property/arbitrary/BuilderTest.kt

```diff
@@ -195,8 +196,8 @@ class BuilderTest : FunSpec() {
             arbitrary(classifier) { arb.bind() }.single() shouldBe 5
             arbitrary(shrinker, classifier) { arb.bind() }.single() shouldBe 5
             arbitrary(listOf(5)) { arb.bind() }.single() shouldBe 5
-            arbitrary({ 5 }) { arb.bind() }.single() shouldBe 5
-            arbitrary({ 5 }, shrinker) { arb.bind() }.single() shouldBe 5
+            arbitrary({ 5.asSample() }) { arb.bind() }.single() shouldBe 5
+            arbitrary({ 5.asSample() }, shrinker) { arb.bind() }.single() shouldBe 5
```

Comment:
@sksamuel the `asSample()` is the part I meant - that looks slightly ugly now. Don't really have a better idea though.

Some thought:

```kotlin
inline fun <reified A : Any> sample(block: () -> A) = block().let {
    when {
        it is Sample<A> -> it
        else -> it.asSample()
    }
}

arbitrary(sample { 5 }, /* ... */)
```